src/schemas/default.ts

```diff
@@ -0,0 +1,41 @@
+import { z } from "zod";
+import { Field } from "../api/types";
+
+export const getMetadataBasedSchema = (
+  fields: Field[],
+  customInputs: {
+    [key: string]: z.ZodTypeAny;
+  }
+) => {
+  const newObj: {
+    [key: string]: z.ZodTypeAny;
+  } = {};
+
+  for (const field of fields) {
+    newObj[field.name] = z.string().optional();
+  }
+
+  for (const key of Object.keys(customInputs)) {
+    newObj[key as keyof typeof newObj] =
+      customInputs[key as keyof typeof newObj];
+  }
+
+  const schema = z
+    .object({
+      ...newObj,
+      WhoId: z.string().min(1).optional(),
```

Comment:
We always need to build the validation schema from sObject metadata as we can't guarantee these fields will exist or be mandatory.

There shouldn't be any circumstance where there are hard coded sObject fields.

(as above)